### PR TITLE
[Sensor] Equirectangular sensor

### PR DESCRIPTION
**Pull request by vauduong**
## Motivation and Context

Given a cubemap, implement an Equirectangular Sensor and shader
Screen Shot 2021-03-15 at 3 48 53 AM

## How Has This Been Tested

Build and test

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by bigbike** [comment]
I see a lot of code duplications between the `Equirectangular` and the `FisheyeSensor`. 
Maybe `Equirectangular` should be sub-classed from `FisheyeSensor`?

**Comment by Skylion007** [comment]
In Habitat-Lab, we had them use a shared base class called CameraProjection.

**Comment by Skylion007** [comment]
https://github.com/facebookresearch/habitat-lab/blob/f89d53ea47e784fdf6e78fadccded645bb5913a7/habitat_baselines/common/obs_transformers.py#L226

**Comment by vauduong** [comment]
> I see a lot of code duplications between the `Equirectangular` and the `FisheyeSensor`.
> Maybe `Equirectangular` should be sub-classed from `FisheyeSensor`?

That's a good point, only the frag shader and the uniforms are different. I think it would make more sense for EquirectangularSensor and FisheyeSensor to be subclassed from `CubeMapSensor` though, since that is the underlying connection. What do you think?

**Comment by bigbike** [comment]
`FisheyeSensor` is already a base class.
Let `EquirectangularSensor` inherit from `FisheyeSensor` (we can think about the name in the future, but leave it like this for now).

**Comment by vauduong** [comment]
> `FisheyeSensor` is already a base class.
> Let `EquirectangularSensor` inherit from `FisheyeSensor` (we can think about the name in the future, but leave it like this for now).

Ok can do! Thanks for the clarification!

File: src/shaders/equirectangular.frag [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+#endif
+
+// ------------ output -----------------------
+#if defined(COLOR_TEXTURE)
+layout(location = OUTPUT_ATTRIBUTE_LOCATION_COLOR) out vec4 fragmentColor;
+#endif
+
+void main(void) {
+  vec3 m;
+  float PI = 3.1415926535897932384626433832795;
```
Comment (Skylion007) [inline src/shaders/equirectangular.frag:25]: Shouldn't this be a const?

**Comment by Skylion007** [comment]
Would appreciate a pass by @mosra at some point to double check we aren't doing anything silly with Magnum and/or Shaders. Also @vauduong your test case .png file is empty which is probably not what you wanted.

**Comment by vauduong** [comment]
Thanks for the catch @Skylion007 ! That's been fixed. 

Making EquirectangularSensor subclass FisheyeSensor will take some more refactoring.

**Comment by mosra** [comment]
I'm confused. I see a lot of extremely similar code between this and #1131. What's the relation of those two? Don't they mutually conflict with each other?

**Comment by bigbike** [comment]
@mosra: #1131 will be checked in first. Then this PR will rebase on top it.

**Comment by Skylion007** [comment]
@vauduong #1131 is merged in. Please rebase.

**Comment by bigbike** [comment]
The `viewer` really needs to be refactored.
Will let #1066 be committed in first.

**Comment by bigbike** [comment]
close it since we have #1179

**Pull request closed without merging**